docs-react-native/react-native/docs/ios/remote-notification-support.md

```diff
@@ -95,79 +76,141 @@ At this point everything should still be running normally. This is the final ste
+## Mutate the content with Notifee:
+> 🔗 [Apple Docs: Modifying content in newly delivered notifications](https://developer.apple.com/documentation/usernotifications/modifying-content-in-newly-delivered-notifications#Implement-your-extensions-handler-methods)
 
-## Edit the payload
-Now everything is setup in your app, you can alter your notification payload in two ways:
+#### Objective C:
 
-1. Update the message payload, sent via your backend
-2. In a Notification Service Extension in your app when a device receives a remote message
+In your `NotifeeNotificationService.m` file you should have method `didReceiveNotificationRequest` where we are calling `NotifeeExtensionHelper`. Now you can modify
+`bestAttemptContent` before you send it to `NotifeeExtensionHelper`:
 
-> Make sure that you will set `mutable-content: 1` (mutableContent if you are using firebase admin sdk) when sending notification otherwise Notification Service Extension will NOT be triggered
+```objectivec
+- (void)didReceiveNotificationRequest:(UNNotificationRequest *)request withContentHandler:(void (^)(UNNotificationContent * _Nonnull))contentHandler {
+    self.contentHandler = contentHandler;
+    self.bestAttemptContent = [request.content mutableCopy];
+
+     // You can also modify the payload as you please here.
+     // You have around 30 seconds
+
+     // Notifee will mutate the notification according to notifee_options
+    [NotifeeExtensionHelper populateNotificationContent:request
+                                withContent: self.bestAttemptContent
+                                withContentHandler:contentHandler];
+}
+```
 
-> Make sure that you will set `content-available: 1` (contentAvailable if you are using firebase admin sdk) if you want to receive notification when your app is in foreground
+#### Swift: 
 
+In your `NotifeeNotificationService.swift` file, use the `didReceive(_:withContentHandler:)` method to enable the `NotifeeExtensionHelper`.
+```swift
+override func didReceive(_ request: UNNotificationRequest,
+                         withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
+    self.contentHandler = contentHandler
+    self.bestAttempt = (request.content.mutableCopy() as? UNMutableNotificationContent)
 
-### 1. Update the message payload, sent via your backend
+    // Here you can also modify the content of the notification as you please here.
+    // You have around 30 seconds
 
-```json
-// FCM
-{
-    notification: {
-      title: 'A notification title!',
-      body: 'A notification body',
+
+    // Notifee will mutate the notification according to notifee_options
+    NotifeeExtensionHelper.populateNotificationContent(request,
+                            with: self.bestAttempt!,
+                            withContentHandler: contentHandler)
+}
+```
+
+<br> 
+
+Before moving to the next step, run the app and check it builds successfully – make sure you have the correct target selected. 
```

Comment:
I also like this suggestion here again - so they do a build test once before they try to do custom code, and now they do another build test after trying to make custom code but before moving on